src/app/loadout-builder/PerkPicker.tsx

```diff
@@ -171,7 +171,7 @@ function mapStateToProps() {
         for (const plugSetHash of sets) {
           const plugSetItems = itemsForPlugSet(profileResponse, plugSetHash);
           for (const plugSetItem of plugSetItems) {
-            if (plugSetItem.canInsert) {
+            if (plugSetItem.canInsert || plugSetItem.insertFailIndexes.length) {
```

Comment:
This feels a little error prone but I don't see an alternative.

Reply:
I think it'll be OK for now, but it requires some documentation to explain the weirdness.

Also, do we need to implement the same fix in `SocketDetails`? Can we refactor a common function to calculate this?

Reply:
I didn't realise we were doing it there too, I will look into it and commonise it somewhere.